static_report/src/components/ComparisonReport/ComparisonReport.tsx

```diff
@@ -354,16 +326,46 @@ function CompareProfile({ base, input }) {
   );
 }
 
-export default function ComparisonReport({ base, input, reportName }) {
+export default function ComparisonReport({
+  data,
+  name: reportName,
+}: {
+  data: ComparisonReportSchema;
+  name: string;
+}) {
   const [testDetail, setTestDetail] = useState(null);
-  const { isOpen, onOpen, onClose } = useDisclosure();
+  const modal = useDisclosure();
+
+  const { base, input } = data;
+  const baseTables = base.tables[reportName];
+  const inputTables = input.tables[reportName];
 
-  const tBase = getComparisonAssertionTests({
-    assertion: base?.assertion_results,
+  const baseOverview = getComparisonAssertionTests({
```

Comment:
same repeating pattern here. 
Maybe a wrapper around  `getComparisonAssertionTests` to execute across all 4 properties?